include/bmi/Bmi_Py_Adapter.hpp

```diff
@@ -248,7 +248,8 @@ namespace models {
                     return "long long";
                 } else if (py_type_name == "longlong" && item_size == sizeof(long long)) {
                     return "long long"; //numpy type
-                } else if (py_type_name == "float" && item_size == sizeof(float)) {
+                } else if ( (py_type_name == "float" || py_type_name == "float32" || py_type_name == "np.float32" ||
+                           py_type_name == "numpy.float32") && item_size == sizeof(float)) {
```

Comment:
We might need to account for this as [`numpy.single`](https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.single) as well: 

```suggestion
                           py_type_name == "numpy.float32" || py_type_name == "np.single" ||  py_type_name == "numpy.single") && item_size == sizeof(float)) {
```